app/controllers/api/v0/admins_controller.rb

```diff
@@ -1,14 +1,19 @@
 module Api
   module V0
     class AdminsController < BaseController
+      before_action :validate_admin
+
       def add_portfolio
         portfolio = Portfolio.create!(portfolio_params)
         render json: portfolio
+      rescue ActiveRecord::RecordInvalid => e
+        render :json => { :errors => e.message }, :status => :unprocessable_entity
       end
 
       def add_portfolio_item_to_portfolio
-        portfolio = Portfolio.find_by(id: params[:portfolio_id])
-        render json: portfolio.add_portfolio_item(params[:portfolio_item_id])
+        portfolio = Portfolio.find(params[:portfolio_id])
+        portfolio_item = PortfolioItem.find(params[:portfolio_item_id])
+        render json: portfolio.add_portfolio_item(portfolio_item)
```

Comment:
Probably also need to render 404 on  `AR::RecordNotFound`

Reply:
@bdunne the 404 is handled directly by rails when it gets the RecordNotFound exception

Reply:
I didn't know that.  I'll have to clean that up for topological_inventory-api